linkml/generators/pydanticgen/array.py

```diff
@@ -366,13 +489,177 @@ def complex_dimensions(self, array: ArrayExpression) -> RangeResult:
         return res
 
 
-class NPTypingArray(ArrayRangeGenerator):
+class NumpydanticArray(ArrayRangeGenerator):
     """
-    Represent array range with nptyping, and serialization/loading with an ArrayProxy
+    Represent array range with :class:`numpydantic.NDArray` annotations,
+    allowing an abstract array specification to be used with many different array
+    libraries.
     """
 
-    REPR = ArrayRepresentation.NPARRAY
+    REPR = ArrayRepresentation.NUMPYDANTIC
+    NUMPYDANTIC_VERSION = "1.2.1"
```

Comment:
semantics on this - should this be the 'target' version or the current numpydantic version if present and installed at generation time? the goal would be to hint at what version of numpydantic this model could be used with, but there's not really a good way of keeping this true.

maybe it should be "minimum numpydantic version," so if it gets updated to use newer version features that's when you would update this. currently we're at `v1.6.1` and that just added `Union` types which we would want to have, so i'll bump this

Reply:
i'm planning on adding linkml tests into the numpydantic tests, that's what the `pydanticgen_npd` mark is for, just fyi, so i'll be testing for not breaking this downstream